task2/src/components/home/MovieCatalog/MovieCatalog.jsx

```diff
@@ -1,18 +1,61 @@
-import { MovieService } from '@containers';
-import { MovieCard } from '@components/home'
+import PropTypes from 'prop-types';
+import { BoldText } from '@components/global';
+import {
+    MovieCard,
+    MovieCatalogCounter
+} from '@components/home';
 import { MovieCatalog as Wrapper } from './MovieCatalog.styles';
 
-//TODO: DELETE IT and replace using CSS FLEX
-const MovieCatalog = () => (
-    <Wrapper>
-        {
-            MovieService
-                .getMovies()
-                .map((movie, idx) =>
-                    <MovieCard key={idx} movie={movie} />
-                )
-        }
-    </Wrapper>
-);
+const MovieCatalog = (props) => {
+
+    //TODO: rewrite it with real API data. Hardcoded :(
+    const activeFilter = props.activeFilter.toLowerCase();
+    const movies = (props.activeFilter === 'ALL' ?
```

Comment:
logic of the active movies should be in the above component

u handle filter or sorting -> set new movies by this filter -> and pass new movies down (instead of active filter)

Reply:
I have a question. So I should use **componentDidUpdate** to filter and sort mock API data? 
And for performance check prevState with newState. 
I can show an example today on the meeting